addon/instance-initializers/ember-app-scheduler.ts

```diff
@@ -0,0 +1,13 @@
+import type ApplicationInstance from '@ember/application/instance'
```

Comment:
I really do not want to add an initializer/instance-initializer to this library...

Reply:
also missing a semi colon

Reply:
I feared a response like that. 😅

I still believe moving to the `RouterService` and destroyable is the right move.

Where would you want to intialize it? `application` `beforeModel()`?

Reply:
> I still believe moving to the RouterService and destroyable is the right move.

Yes, totally agree.

> Where would you want to intialize it? application beforeModel()?

Hmm, not sure. The only thing that reliably exists for _every_ app (and happens to be properly aligned with the application lifecycle we care about) is `app/router.js`. We could leverage destroyables and still ask folks to use `setupRouter` like we do now...

That still lets us kill the `reset` stuff which is nice...

Reply:
`application` `beforeModel()` probably also wouldn't work reliably, as `beforeModel()` would get called after the initial `routeWillChange`.

I really like the pain-free "hands off" approach of using an initializer, but I understand, that initializers should be avoided. Do you think an optional initializer (with a flag in `config/environment.js` defaulting to `true`) is also bad? It would allow the addon to "just work" after installation, but gives users the freedom to resort back to manual setup, where needed.

Essentially `setupRouter(router: RouterService | Router)` could stay with that signature then and for users preferring the manual setup, we would recommend either calling it in `Router#init()` or in `Application#buildInstance()`.

Or what would you propose?

I'd really like to move off of the Ember `Router` entirely to just the `RouterService`, but if the setup is put in the Ember `Router` _anyway_, it doesn't _really_ make sense. 😅

Reply:
> application beforeModel() probably also wouldn't work reliably, as beforeModel() would get called after the initial routeWillChange.

Yes, doing the work from within a hook that is itself part of a transition is not going to work.

> I'd really like to move off of the Ember Router entirely to just the RouterService, but if the setup is put in the Ember Router anyway, it doesn't really make sense. 😅

I don't think we actually support _not_ defining an `app/router.js`, and in general that file is very commonly used/modified by users. I think doing a bunch of gymnastics to avoid any modification there is a bit odd.

>  It would allow the addon to "just work" after installation, but gives users the freedom to resort back to manual setup, where needed.

FWIW, there are other ways to enable the "it just works" sort of thing. For example, we could add a blueprint that does the mutation for you.

Reply:
Also, IMHO, either way the reset functionality should be dropped in favor of `registerDestructor` from within `setupRouter`.

Reply:
Agree with this. This is a good pattern to leverage, as it effectively sets up and registers teardown in one place.

Reply:
I've removed the initializer for now.